packages/daemon/src/host.js

```diff
@@ -203,24 +204,34 @@ export const makeHostMaker = ({
       if (resultName !== undefined) {
         assertPetName(resultName);
       }
-      if (petNames.length !== codeNames.length) {
+      if (petNamePaths.length !== codeNames.length) {
         throw new Error('Evaluator requires one pet name for each code name');
       }
 
       const formulaIdentifiers = await Promise.all(
-        petNames.map(async (petName, index) => {
+        petNamePaths.map(async (petNameOrPath, index) => {
           if (typeof codeNames[index] !== 'string') {
             throw new Error(`Invalid endowment name: ${q(codeNames[index])}`);
           }
-          const formulaIdentifier = lookupFormulaIdentifierForName(petName);
-          if (formulaIdentifier === undefined) {
-            throw new Error(`Unknown pet name ${q(petName)}`);
+
+          const petNamePath = petNamePathFrom(petNameOrPath);
+          if (petNamePath.length === 1) {
+            const formulaIdentifier = lookupFormulaIdentifierForName(
+              petNamePath[0],
+            );
+            if (formulaIdentifier === undefined) {
+              throw new Error(`Unknown pet name ${q(petNamePath[0])}`);
+            }
+            return formulaIdentifier;
           }
-          return formulaIdentifier;
+
+          const { formulaIdentifier: lookupFormulaIdentifier } =
+            await provideLookupFormula(petNamePath);
+          return lookupFormulaIdentifier;
```

Comment:
Would be good to factor out a `lookupFormulaIdentifierForPath` function. This won’t be the last place we use it.

Reply:
Decision: We will do this later if necessary.